docs/tutorials/gallery_adding_components.py

```diff
@@ -0,0 +1,125 @@
+"""
+# Adding Components to Ragna
+
+Ragna has builtin support for several [assistants][ragna.assistants] and 
+[source storages][ragna.source_storages], but there may be cases where you 
+want to use one that is not currently supported.
+
+This tutorial walks you through the basics of adding components
+that are not currently officially supported. 
+"""
+
+# %%
+# ## Adding an LLM Assistant
+
+from typing import Iterator
+
+from ragna.core import Assistant, Source
+
+
+class TutorialAssistant(Assistant):
```

Comment:
Before we actually put the code, we need to explain what the `TutorialAssistant` will do. Something along the lines of "for simplicity, we are not going to implement an actual LLM here, but rather a demo assistant that just mirrors back the inputs. This is similar as the RagnaDemoAssistant".